scripts/create-nbsphinx-links.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+shopt -s globstar nullglob
+
+# get all notebooks
+for nb_path in demos/**/*.ipynb; do
+  nb_link="docs/${nb_path%.ipynb}.nblink"
+
+  # recurse through parent directories and create index.txt
+  nb_link_dir="$(dirname "$nb_link")"
+  while [ "$nb_link_dir" != "docs" ]; do
+    nb_link_parent_index="$nb_link_dir/index.txt"
+    if [[ ! -f "$nb_link_parent_index" ]]; then
+      echo "Creating '$nb_link_parent_index'"
+      mkdir -p "$nb_link_dir"
+      cat > "$nb_link_parent_index" << EOF
+$(basename "$nb_link_dir")
```

Comment:
Maybe just this can be capitalised? (ref my comment earlier in the PR?)

Reply:
Hmm yeah that's a reasonable idea! Seems like every directory is kebab case, except for `community_detection` - I guess I could put a formatting rule in here to replace `-`/`_` with space, and capitalise each word?

Reply:
I like the concept, but figuring out the sed is tricky!  Possibly awk more suited?

Reply:
BSD sed doesn't support `\U` to capitalise, so I'm trying something with `awk` now..